### ISSUE TITLE
[3.x] `_input` reports `is_action_just_pressed()` multiple times per frame 

### ISSUE THREAD

**Issue by lawnjelly** [godotengine/godot#97526]
### Tested versions

Reproducible in 3.6 stable
Not reproducible in 3.5.3

### System information

All

### Issue description

If `_input()` is called multiple times per frame, then `is_action_just_pressed("some_action")` will return true every time until the next frame.

I'm not yet sure whether this affects other actions aside from mouse wheel.

UPDATE:
* It does not occur with regular mouse buttons for me, only mouse wheel
* Doesn't occur for key presses

It seems to occur with mousewheel because this creates a `press` followed by `release` in rapid succession, whereas regular mouse click is unlikely to press and release on the same frame.

I saw reported on reddit:
https://www.reddit.com/r/godot/comments/1fgi328/36_registers_mouse_wheel_event_more_than_once_35/
But wanted to create an issue for reference.

The change in behaviour is introduced by my PR #77040 which fixes #73339 .

This is a similar confusion as #80158 in master (which likely would also occur in 3.x).

### Steps to reproduce

```
func _input(_event):
	print("input frame: ", Engine.get_idle_frames())
	
	if Input.is_action_just_pressed("wheel_up"):
		print("wheel_up")
	if Input.is_action_just_pressed("wheel_down"):
		print("wheel_down")
```

Results in 3.6:
```
input frame: 6
input frame: 7
input frame: 42
wheel_down
input frame: 42
wheel_down
input frame: 57
wheel_down
input frame: 57
wheel_down
input frame: 197
input frame: 198
input frame: 199
```
Results in 3.5.3:
```
input frame: 6
input frame: 7
input frame: 42
wheel_down
input frame: 42
input frame: 57
wheel_down
input frame: 57
input frame: 197
input frame: 198
```
i.e. the `is_action_just_pressed()` would only occur once.

### Minimal reproduction project (MRP)

[Double_Input.zip](https://github.com/user-attachments/files/17160265/Double_Input.zip)

## Discussion

I am hesitating to call this a "bug", but it is a change in behaviour.

To some extent this has always been the case, because if multiple `_input()` come in on a single frame for any action that is just pressed, it will, in most cases report the same action as true multiple times. That is correct, because any action pressed *IS* considered pressed throughout the frame. 

What did happen previously though was an anomaly: on a mouse button press the press created the input with the action just pressed, and the depress created an input with the action just pressed set to false (on the same frame).

This resulted in "expected" behaviour for handling input via `_input()`, however created the bug #73339 . Fixing the bug can result in less "expected" behaviour via `_input()`, and can make input handling more complex in this case.

To make life easier we could look at an option to simply auto-cancel the action once it had been read once with `is_action_just_pressed()`, which would be friendly to both solutions to reading input.

**Comment by timothyqiu** [comment]
I don't think this is a bug.

This code snippet is based on a misunderstanding of how `Input` is related to the input event system. It only "happened" to work in previous versions. Your PR exposed the bug in that code snippet instead of creating it.

**Comment by lawnjelly** [comment]
> I don't think this is a bug.

> This code snippet is based on a misunderstanding of how Input is related to the input event system. It only "happened" to work in previous versions. Your PR exposed the bug in that code snippet instead of creating it.

Absolutely, this is why I haven't labelled it as bug. It probably was occurring already in some situations (like pressing two keys on the same frame and handling in `_input()`) but users weren't aware of it.

However it is something which I suspect we could make easier for users to handle.

I am just preparing a draft PR which could potentially deal with this, simply by auto-cancelling the action after it has been read once in `is_action_just_pressed()`. This would seem to cover the vast majority of cases.

My only question is whether to make this optional (for backward compatibility, in case users are excepting actions to return true multiple times). Can discuss on PR though.

**Comment by timothyqiu** [comment]
I made a proposal about making `Input.is_action_just_pressed()` equivalent to `event.is_action_pressed()` in these input event callbacks: https://github.com/godotengine/godot-proposals/issues/10843

**Comment by lawnjelly** [comment]
I've closed my PR for retiring, as @timothyqiu points it is already possible to work around by using `event.is_action_pressed()` within `_input()`.

I'm not so familiar with using `_input()` callback for anything but mouse look, so apologies.

So this likely becomes a documentation issue, or a case for https://github.com/godotengine/godot-proposals/issues/10843 .

**Comment by lawnjelly** [comment]
Closing as hopefully the documentation and warning should help prevent this problem.
